app/routes/jobs/job.py

```diff
@@ -0,0 +1,79 @@
+"""Jobs represent the steps performed during asset creation.
+
+You can view a single tasks or all tasks associated with as specific
+asset. Only _service accounts_ can create or update tasks.
+"""
+import json
+from typing import Any, Dict
+from uuid import UUID
+
+import botocore
+from fastapi import APIRouter, HTTPException, Path
+from fastapi.responses import ORJSONResponse
+
+from ...models.pydantic.user_job import UserJob, UserJobResponse
+from ...settings.globals import STATE_MACHINE_ARN
+from ...utils.aws import get_sfn_client
+
+router = APIRouter()
+
+
+@router.get(
+    "/{job_id}",
+    response_class=ORJSONResponse,
+    tags=["Jobs"],
+    response_model=UserJobResponse,
+)
+async def get_job(*, job_id: UUID = Path(...)) -> UserJobResponse:
+    """Get job status.
+
+    Jobs expire after 90 days.
+    """
+    try:
+        job = await _get_user_job(job_id)
+        return UserJobResponse(data=job)
+    except botocore.exceptions.ClientError as e:
+        raise HTTPException(status_code=404, detail=str(e))
+
+
+async def _get_user_job(job_id: UUID) -> UserJob:
+    execution = _get_sfn_execution(job_id)
+
+    if execution["status"] == "SUCCEEDED":
+        output = json.loads(execution["output"])
+
+        return UserJob(
+            job_id=job_id,
+            status="saved",
+            download_link=output["download_link"],
```

Comment:
I'll update the step function documentation to indicate that the aggregation function doesn't need to do a PATCH call to /job/{job_id}, since you're getting the result info directly from the step function.